diff --git a/lsp/leanls.lua b/lsp/leanls.lua
index 8d7191bc..803def1b 100644
--- a/lsp/leanls.lua
+++ b/lsp/leanls.lua
@@ -221,9 +221,9 @@ return {
     -- fall back to `lean --server` for core lean4 or standalone files.
     local local_cmd
     if cmd_cwd and will_use_lake_server(cmd_cwd) then
-      local_cmd = { 'lake', 'serve', '--', config.root_dir }
+      local_cmd = { '@lean4@/bin/lake', 'serve', '--', config.root_dir }
     else
-      local_cmd = { 'lean', '--server', config.root_dir }
+      local_cmd = { '@lean4@/bin/lean', '--server', config.root_dir }
     end
     return vim.lsp.rpc.start(local_cmd, dispatchers, {
       cwd = cmd_cwd,
