diff --git a/lua/lspconfig/configs/rust_analyzer.lua b/lua/lspconfig/configs/rust_analyzer.lua
index b89546a..9367fc3 100644
--- a/lua/lspconfig/configs/rust_analyzer.lua
+++ b/lua/lspconfig/configs/rust_analyzer.lua
@@ -48,7 +48,7 @@ return {

       if cargo_crate_dir ~= nil then
         local cmd = {
-          'cargo',
+          '@cargo@/bin/cargo',
           'metadata',
           '--no-deps',
           '--format-version',
