diff --git a/lua/snacks/image/convert.lua b/lua/snacks/image/convert.lua
index ddc1179..d81ec99 100644
--- a/lua/snacks/image/convert.lua
+++ b/lua/snacks/image/convert.lua
@@ -67,7 +67,7 @@ local commands = {
     ft = "pdf",
     cmd = {
       {
-        cmd = "typst",
+        cmd = "@typst@/bin/typst",
         args = { "compile", "--format", "pdf", "--pages", 1, "{src}", "{file}" },
       },
     },
@@ -81,7 +81,7 @@ local commands = {
     cmd = {
       {
         cwd = "{dirname}",
-        cmd = "tectonic",
+        cmd = "@tectonic@/bin/tectonic",
         args = { "-Z", "continue-on-errors", "--outdir", "{cache}", "{src}" },
       },
       {
@@ -119,7 +119,7 @@ local commands = {
     end,
     cmd = {
       {
-        cmd = "magick",
+        cmd = "@imagemagick@/bin/magick",
         args = { "identify", "-format", "%m %[fx:w]x%[fx:h] %xx%y", "{src}[0]" },
       },
       {
@@ -176,7 +176,7 @@ local commands = {
 
       vim.list_extend(args, { "-write", "{file}", "-identify", "-format", "%m %[fx:w]x%[fx:h] %xx%y", "{file}.info" })
       return {
-        { cmd = "magick", args = args },
+        { cmd = "@imagemagick@/bin/magick", args = args },
         not Snacks.util.is_win and { cmd = "convert", args = args } or nil,
       }
     end,
